test/controllers/projects/samples/clones_controller_test.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module Projects
+  module Samples
+    class ClonesControllerTest < ActionDispatch::IntegrationTest
+      setup do
+        sign_in users(:john_doe)
+        @namespace = groups(:group_one)
+        @project = projects(:project1)
+        @new_project = projects(:project2)
+        @sample1 = samples(:sample1)
+        @sample2 = samples(:sample2)
+      end
+
+      test 'not clone samples with empty params' do
+        post namespace_project_samples_clone_path(@namespace, @project, format: :turbo_stream),
+             params: {
+               clone: {
+                 new_project_id: nil,
+                 sample_ids: nil
+               }
+             }
+        assert_response :unprocessable_entity
+      end
+
+      test 'not clone samples with no sample ids' do
+        post namespace_project_samples_clone_path(@namespace, @project, format: :turbo_stream),
+             params: {
+               clone: {
+                 new_project_id: @new_project.id,
+                 sample_ids: []
+               }
+             }, as: :json
+        assert_response :unprocessable_entity
+      end
+
+      test 'not clone samples with into same project' do
+        post namespace_project_samples_clone_path(@namespace, @project, format: :turbo_stream),
+             params: {
+               clone: {
+                 new_project_id: @project.id,
+                 sample_ids: [@sample1.id, @sample2.id]
+               }
+             }
+        assert_response :unprocessable_entity
+      end
+
+      test 'not clone samples with same sample name' do
```

Comment:
Once the controller comment is addressed this will need to be updated as well as it shouldn't throw a partial content status if only one sample id was provided and that sample fails to clone

Reply:
5aa4b43e9c981feeaff46aab2d28b14e1179ba36